### PR TITLE
Add 'other errors' counter to Retry

### LINKED ISSUE #1746: Add `other` counter to Retry for fully fine-grained retry configuration
```python
from urllib3 import PoolManager                                                   
from urllib3.util.retry import Retry                                              
from urllib3.util.timeout import Timeout                                          
                                                                                  
                                                                                  
def main(**kwargs):                                                               
    url = 'https://78.155.216.172'                                                
    mgr = PoolManager()                                                           
    retry_opts = {                                                                
        'total': None,                                                            
        'redirect': 3,                                                            
        'raise_on_redirect': True,                                                
        'connect': 1,                                                             
        'read': 1,                                                                
    }                                                                             
    res = mgr.request(                                                            
        'GET',                                                                    
        url,                                                                      
        retries=Retry(                                                            
            **retry_opts,                                                         
        ),                                                                        
        timeout=Timeout(                                                          
            connect=3,                                                            
            read=5,                                                               
        ),                                                                        
    )
```
Retry option `connect=1` is ignored. Urllib tries to connect again and again.
Console output:
```
Starting new HTTPS connection (1): 78.155.216.172:443
Incremented Retry for (url='/'): Retry(total=None, connect=1, read=1, redirect=3, status=None)
Retrying (Retry(total=None, connect=1, read=1, redirect=3, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)'),)': /
Starting new HTTPS connection (2): 78.155.216.172:443
Incremented Retry for (url='/'): Retry(total=None, connect=1, read=1, redirect=3, status=None)
Retrying (Retry(total=None, connect=1, read=1, redirect=3, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)'),)': /
Starting new HTTPS connection (3): 78.155.216.172:443
Incremented Retry for (url='/'): Retry(total=None, connect=1, read=1, redirect=3, status=None)
Retrying (Retry(total=None, connect=1, read=1, redirect=3, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)'),)': /
Starting new HTTPS connection (4): 78.155.216.172:443
Incremented Retry for (url='/'): Retry(total=None, connect=1, read=1, redirect=3, status=None)
Retrying (Retry(total=None, connect=1, read=1, redirect=3, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)'),)': /
Starting new HTTPS connection (5): 78.155.216.172:443
Incremented Retry for (url='/'): Retry(total=None, connect=1, read=1, redirect=3, status=None)
Retrying (Retry(total=None, connect=1, read=1, redirect=3, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)'),)': /
Starting new HTTPS connection (6): 78.155.216.172:443

... AND SO ON
```

### PR DESCRIPTION
**Pull request by hodbn**
Closes #1746

### CODE REVIEW

**Comment by pquentin** [comment]
@hodbn Can you please rebase this on top of master? (Or merge master if your prefer) We'll get coverage info that way.

File: src/urllib3/util/retry.py [hunk 1]
```diff
@@ -95,6 +95,15 @@ class Retry(object):
 
         Set to ``0`` to fail on the first retry of this type.
 
+    :param int other:
+        How many times to retry on other errors.
+
+        Other errors are errors that are not connect, read, redirect or status errors.
+        These errors might be raised after the request was sent to the server, so the
+        request might have side-effects.
+
```
Comment (pquentin) [inline src/urllib3/util/retry.py:104]: For `total`, we say:

> It's a good idea to set this to some sensibly-high value to account for unexpected edge cases and avoid infinite retry loops.

However, with `other`, I believe the situation changes. I'd remove this line from total and add this line here:

> It's a good idea to set this to 0 to account for unexpected edge cases and avoid infinite retry loops.

What do you think?

Comment (hodbn) [reply src/urllib3/util/retry.py:104]: You are right :)
I've added the prefix:
> If ``total`` is not set,

**Comment by pquentin** [comment]
Closing/reopening to get the Sphinx fix.

**Comment by hodbn** [comment]
@pquentin can you take a look at this? :)

**Comment by pquentin** [comment]
Yes, a review is long overdue, sorry about that. If you don't get any before Monday, I'll try to take a look.

**Pull request merged**